inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp

```diff
@@ -65,10 +67,11 @@ void MKLDNNPadNode::initSupportedPrimitiveDescriptors() {
     if (!supportedPrimitiveDescriptors.empty())
         return;
 
-    InferenceEngine::Precision precision = InferenceEngine::Precision::FP32;
+    InferenceEngine::Precision precision = getCnnLayer()->insData[0].lock()->getPrecision();
```

Comment:
This change is not always valid: for example your implementation doesn't support BIN precision. You need to create list of supported by your implementation precisions and check if getCnnLayer()->insData[0].lock()->getPrecision() is present in the list. Otherwise you need to create supportedPrimitiveDescriptors for FP32 precision in case getCnnLayer()->insData[0].lock()->getPrecision().is_float() == true and I32 if false. The plug-in will automatically do all necessary data type conversions.

Reply:
Its my mistake. Thank you! Fixed

Reply:
BTW, implementation limitation on precisions is true only for Constant mode with non zero const value.  It seems worthwhile to check precisions only for this case.